@Timed annotation does not work with Graal

Micronaut version: 1.1.0.RC2
Graal: 1.0.0-rc14, installed with sdkman

Notes:
1. binders.jvm.enabled is set to false to work with Graal
2. Reproducer available at: https://github.com/dekstroza/sandbox.git branch working-mongonaut

To reproduce:
```
git clone git@github.com:dekstroza/sandbox.git
git checkout -b working-mongonaut
cd mongonaut
./mvnw clean install #(or just mvn clean install)
```

Having prometheus and metrics enabled on the following:

```
@Singleton
@Controller("/mongonaut")
public class AlarmController {

    private final MongoClient mongoClient;

    public AlarmController(MongoClient mongoClient) {
        this.mongoClient = mongoClient;
    }

    @Timed(value = "method.alarms.getall",percentiles = {0.5, 0.95},histogram = true, description = "Read all alarms timer")
    @Get(value = "/alarms")
    public Flowable<Alarm> getAll() {
        return Flowable.fromPublisher(getCollection().find());

    }

    @Timed(value = "method.alarms.save", percentiles = {0.5, 0.95},histogram = true, description = "Save alarm timer")
    @Post("/alarms")
    public Single<Alarm> save(Integer id, String name, String severity) {
        Alarm alarm = new Alarm(id, name, severity);
        return Single.fromPublisher(getCollection().insertOne(alarm)).map(success -> alarm);
    }

    private MongoCollection<Alarm> getCollection() {
        return mongoClient.getDatabase("mongonaut").getCollection("alarms", Alarm.class);
    }
```
prometheus metrics is correctly exposed on /prometheus when running 

```
java -jar target/mongonaut-1.0.0-SNAPSHOT.jar
```
but not when compiled with Graal. 
In this case, prometheus endpoint is still exposed and contains the usual metrics but  the "@Timed" one is not present at all. Application has no errors in the logs.

Image is created using following Dockerfile
```
FROM oracle/graalvm-ce:1.0.0-rc14 as graalvm
ARG JAR_FILE
ADD /target/${JAR_FILE} /home/app/mongonaut.jar
WORKDIR /home/app
RUN native-image --no-server --class-path /home/app/mongonaut.jar
FROM frolvlad/alpine-glibc
EXPOSE 8080
COPY --from=graalvm /home/app/mongonaut .
ENTRYPOINT ["./mongonaut"]
```

@graemerocher Did bit of digging and I think this is where the problem is.
TimedInterceptor is getting called, but the stopTimed method is throwing exception which is swallowed insde the try/catch statement. If I add printStackTrace to it, this is the exception. Will check now what is causing it. Also, I have re-built from source micronaut-micrometer and updated it's dependencies towards RC2, fixed few failing tests and tried that as well. Same exception happens, at the same spot.

```
java.lang.IllegalArgumentException: java.lang.NoSuchMethodException: org.HdrHistogram.Histogram.<init>long, long, int
	at org.HdrHistogram.DoubleHistogram.<init>(DoubleHistogram.java:219)
	at org.HdrHistogram.DoubleHistogram.<init>(DoubleHistogram.java:141)
	at org.HdrHistogram.DoubleHistogram.<init>(DoubleHistogram.java:77)
	at io.micrometer.core.instrument.distribution.TimeWindowPercentileHistogram.<init>(TimeWindowPercentileHistogram.java:38)
	at io.micrometer.core.instrument.AbstractTimer.<init>(AbstractTimer.java:85)
	at io.micrometer.prometheus.PrometheusTimer.<init>(PrometheusTimer.java:40)
	at io.micrometer.prometheus.PrometheusMeterRegistry.newTimer(PrometheusMeterRegistry.java:166)
	at io.micrometer.core.instrument.MeterRegistry.lambda$timer$2(MeterRegistry.java:271)
	at io.micrometer.core.instrument.MeterRegistry.getOrCreateMeter(MeterRegistry.java:580)
	at io.micrometer.core.instrument.MeterRegistry.registerMeterIfNecessary(MeterRegistry.java:543)
	at io.micrometer.core.instrument.MeterRegistry.timer(MeterRegistry.java:269)
	at io.micrometer.core.instrument.Timer$Builder.register(Timer.java:464)
	at io.micrometer.core.instrument.composite.CompositeTimer.registerNewMeter(CompositeTimer.java:140)
	at io.micrometer.core.instrument.composite.CompositeTimer.registerNewMeter(CompositeTimer.java:31)
	at io.micrometer.core.instrument.composite.AbstractCompositeMeter.add(AbstractCompositeMeter.java:66)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at java.util.Collections$SetFromMap.forEach(Collections.java:5476)
	at io.micrometer.core.instrument.composite.CompositeMeterRegistry.lambda$null$0(CompositeMeterRegistry.java:65)
	at io.micrometer.core.instrument.composite.CompositeMeterRegistry.lock(CompositeMeterRegistry.java:184)
	at io.micrometer.core.instrument.composite.CompositeMeterRegistry.lambda$new$1(CompositeMeterRegistry.java:65)
	at io.micrometer.core.instrument.MeterRegistry.getOrCreateMeter(MeterRegistry.java:590)
	at io.micrometer.core.instrument.MeterRegistry.registerMeterIfNecessary(MeterRegistry.java:543)
	at io.micrometer.core.instrument.MeterRegistry.timer(MeterRegistry.java:269)
	at io.micrometer.core.instrument.Timer$Builder.register(Timer.java:464)
	at io.micronaut.configuration.metrics.micrometer.intercept.TimedInterceptor.stopTimed(TimedInterceptor.java:182)
	at io.micronaut.configuration.metrics.micrometer.intercept.TimedInterceptor.lambda$intercept$2(TimedInterceptor.java:102)
	at io.reactivex.internal.operators.single.SingleDoOnSuccess$DoOnSuccess.onSuccess(SingleDoOnSuccess.java:54)
	at io.micronaut.reactive.rxjava2.RxInstrumentedSingleObserver.lambda$onSuccess$1(RxInstrumentedSingleObserver.java:80)
	at io.micronaut.http.context.ServerRequestContext.with(ServerRequestContext.java:52)
	at io.micronaut.http.context.ServerRequestContext.lambda$instrument$0(ServerRequestContext.java:68)
	at io.micronaut.reactive.rxjava2.RxInstrumentedSingleObserver.onSuccess(RxInstrumentedSingleObserver.java:84)
	at io.reactivex.internal.operators.single.SingleDoOnError$DoOnError.onSuccess(SingleDoOnError.java:52)
	at io.micronaut.reactive.rxjava2.RxInstrumentedSingleObserver.lambda$onSuccess$1(RxInstrumentedSingleObserver.java:80)
	at io.micronaut.http.context.ServerRequestContext.with(ServerRequestContext.java:52)
	at io.micronaut.http.context.ServerRequestContext.lambda$instrument$0(ServerRequestContext.java:68)
	at io.micronaut.reactive.rxjava2.RxInstrumentedSingleObserver.onSuccess(RxInstrumentedSingleObserver.java:84)
	at io.reactivex.internal.operators.single.SingleDoOnSubscribe$DoOnSubscribeSingleObserver.onSuccess(SingleDoOnSubscribe.java:77)
	at io.micronaut.reactive.rxjava2.RxInstrumentedSingleObserver.lambda$onSuccess$1(RxInstrumentedSingleObserver.java:80)
	at io.micronaut.http.context.ServerRequestContext.with(ServerRequestContext.java:52)
	at io.micronaut.http.context.ServerRequestContext.lambda$instrument$0(ServerRequestContext.java:68)
	at io.micronaut.reactive.rxjava2.RxInstrumentedSingleObserver.onSuccess(RxInstrumentedSingleObserver.java:84)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:64)
	at io.micronaut.reactive.rxjava2.RxInstrumentedSingleObserver.lambda$onSuccess$1(RxInstrumentedSingleObserver.java:80)
	at io.micronaut.http.context.ServerRequestContext.with(ServerRequestContext.java:52)
	at io.micronaut.http.context.ServerRequestContext.lambda$instrument$0(ServerRequestContext.java:68)
	at io.micronaut.reactive.rxjava2.RxInstrumentedSingleObserver.onSuccess(RxInstrumentedSingleObserver.java:84)
	at io.reactivex.internal.operators.single.SingleFromPublisher$ToSingleObserver.onComplete(SingleFromPublisher.java:101)
	at com.mongodb.reactivestreams.client.internal.ObservableToPublisher$1.onComplete(ObservableToPublisher.java:78)
	at com.mongodb.async.client.AbstractSubscription.onComplete(AbstractSubscription.java:147)
	at com.mongodb.async.client.AbstractSubscription.processResultsQueue(AbstractSubscription.java:213)
	at com.mongodb.async.client.AbstractSubscription.tryProcessResultsQueue(AbstractSubscription.java:161)
	at com.mongodb.async.client.AbstractSubscription.addToQueue(AbstractSubscription.java:112)
	at com.mongodb.async.client.SingleResultCallbackSubscription$1.onResult(SingleResultCallbackSubscription.java:47)
	at com.mongodb.reactivestreams.client.internal.PublisherHelper$1.onResult(PublisherHelper.java:36)
	at com.mongodb.reactivestreams.client.internal.PublisherHelper$1.onResult(PublisherHelper.java:33)
	at com.mongodb.async.client.MongoCollectionImpl$1.onResult(MongoCollectionImpl.java:493)
	at com.mongodb.async.client.MongoCollectionImpl$1.onResult(MongoCollectionImpl.java:490)
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:1042)
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:1026)
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49)
	at com.mongodb.async.client.OperationExecutorImpl$2$1.onResult(OperationExecutorImpl.java:117)
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49)
	at com.mongodb.operation.OperationHelper$ConnectionReleasingWrappedCallback.onResult(OperationHelper.java:379)
	at com.mongodb.operation.MixedBulkWriteOperation.addBatchResult(MixedBulkWriteOperation.java:485)
	at com.mongodb.operation.MixedBulkWriteOperation.access$1500(MixedBulkWriteOperation.java:67)
	at com.mongodb.operation.MixedBulkWriteOperation$6.onResult(MixedBulkWriteOperation.java:468)
	at com.mongodb.operation.MixedBulkWriteOperation$6.onResult(MixedBulkWriteOperation.java:443)
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49)
	at com.mongodb.internal.connection.DefaultServer$DefaultServerProtocolExecutor$2.onResult(DefaultServer.java:227)
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49)
	at com.mongodb.internal.connection.CommandProtocolImpl$1.onResult(CommandProtocolImpl.java:85)
	at com.mongodb.internal.connection.DefaultConnectionPool$PooledConnection$1.onResult(DefaultConnectionPool.java:461)
	at com.mongodb.internal.connection.UsageTrackingInternalConnection$2.onResult(UsageTrackingInternalConnection.java:111)
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49)
	at com.mongodb.internal.connection.InternalStreamConnection$2$1.onResult(InternalStreamConnection.java:379)
	at com.mongodb.internal.connection.InternalStreamConnection$2$1.onResult(InternalStreamConnection.java:356)
	at com.mongodb.internal.connection.InternalStreamConnection$MessageHeaderCallback$MessageCallback.onResult(InternalStreamConnection.java:651)
	at com.mongodb.internal.connection.InternalStreamConnection$MessageHeaderCallback$MessageCallback.onResult(InternalStreamConnection.java:618)
	at com.mongodb.internal.connection.InternalStreamConnection$5.completed(InternalStreamConnection.java:494)
	at com.mongodb.internal.connection.InternalStreamConnection$5.completed(InternalStreamConnection.java:491)
	at com.mongodb.internal.connection.AsynchronousSocketChannelStream$BasicCompletionHandler.completed(AsynchronousSocketChannelStream.java:237)
	at com.mongodb.internal.connection.AsynchronousSocketChannelStream$BasicCompletionHandler.completed(AsynchronousSocketChannelStream.java:220)
	at sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:126)
	at sun.nio.ch.Invoker.invokeDirect(Invoker.java:157)
	at sun.nio.ch.UnixAsynchronousSocketChannelImpl.implRead(UnixAsynchronousSocketChannelImpl.java:553)
	at sun.nio.ch.AsynchronousSocketChannelImpl.read(AsynchronousSocketChannelImpl.java:276)
	at sun.nio.ch.AsynchronousSocketChannelImpl.read(AsynchronousSocketChannelImpl.java:297)
	at com.mongodb.internal.connection.AsynchronousSocketChannelStream.readAsync(AsynchronousSocketChannelStream.java:132)
	at com.mongodb.internal.connection.InternalStreamConnection.readAsync(InternalStreamConnection.java:491)
	at com.mongodb.internal.connection.InternalStreamConnection.access$1000(InternalStreamConnection.java:74)
	at com.mongodb.internal.connection.InternalStreamConnection$MessageHeaderCallback.onResult(InternalStreamConnection.java:608)
	at com.mongodb.internal.connection.InternalStreamConnection$MessageHeaderCallback.onResult(InternalStreamConnection.java:593)
	at com.mongodb.internal.connection.InternalStreamConnection$5.completed(InternalStreamConnection.java:494)
	at com.mongodb.internal.connection.InternalStreamConnection$5.completed(InternalStreamConnection.java:491)
	at com.mongodb.internal.connection.AsynchronousSocketChannelStream$BasicCompletionHandler.completed(AsynchronousSocketChannelStream.java:237)
	at com.mongodb.internal.connection.AsynchronousSocketChannelStream$BasicCompletionHandler.completed(AsynchronousSocketChannelStream.java:220)
	at sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:126)
	at sun.nio.ch.Invoker$2.run(Invoker.java:218)
	at sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:473)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
Caused by: java.lang.NoSuchMethodException: org.HdrHistogram.Histogram.<init>long, long, int
	at java.lang.Class.getConstructor0(DynamicHub.java:3082)
	at java.lang.Class.getConstructor(DynamicHub.java:1825)
	at org.HdrHistogram.DoubleHistogram.<init>(DoubleHistogram.java:178)
	... 103 more
```

--- a/micrometer-core/src/main/java/io/micronaut/configuration/metrics/management/endpoint/MetricsEndpoint.java
+++ b/micrometer-core/src/main/java/io/micronaut/configuration/metrics/management/endpoint/MetricsEndpoint.java
@@ -22,6 +22,7 @@ import io.micrometer.core.instrument.Tag;
 import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
 import io.micronaut.configuration.metrics.annotation.RequiresMetrics;
 import io.micronaut.context.ApplicationContext;
+import io.micronaut.core.annotation.TypeHint;
 import io.micronaut.core.bind.exceptions.UnsatisfiedArgumentException;
 import io.micronaut.core.type.Argument;
 import io.micronaut.management.endpoint.annotation.Endpoint;
@@ -52,6 +53,11 @@ import java.util.stream.Collectors;
  */
 @Endpoint(value = MetricsEndpoint.NAME, defaultSensitive = MetricsEndpoint.DEFAULT_SENSITIVE)
 @RequiresMetrics
+@TypeHint(value = {io.micronaut.configuration.metrics.management.endpoint.MetricsEndpoint.class,
+           io.micronaut.configuration.metrics.management.endpoint.MetricsEndpoint.MetricNames.class,
+           io.micronaut.configuration.metrics.management.endpoint.MetricsEndpoint.MetricDetails.class,
+           io.micronaut.configuration.metrics.management.endpoint.MetricsEndpoint.AvailableTag.class,
+           io.micronaut.configuration.metrics.management.endpoint.MetricsEndpoint.Sample.class })
 public class MetricsEndpoint {
 
     /**

--- a/micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java
+++ b/micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java
@@ -45,7 +45,7 @@ import java.util.function.BiConsumer;
  */
 @Singleton
 @RequiresMetrics
-@TypeHint(value = {org.HdrHistogram.Histogram.class,org.HdrHistogram.ConcurrentHistogram.class})
+@TypeHint(value = {org.HdrHistogram.Histogram.class, org.HdrHistogram.ConcurrentHistogram.class})
 public class TimedInterceptor implements MethodInterceptor<Object, Object> {
 
     /**

--- a/micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java
+++ b/micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java
@@ -23,6 +23,7 @@ import io.micronaut.aop.MethodInterceptor;
 import io.micronaut.aop.MethodInvocationContext;
 import io.micronaut.configuration.metrics.annotation.RequiresMetrics;
 import io.micronaut.core.annotation.AnnotationMetadata;
+import io.micronaut.core.annotation.TypeHint;
 import io.micronaut.core.async.publisher.Publishers;
 import io.micronaut.core.util.StringUtils;
 import io.reactivex.Flowable;
@@ -44,6 +45,7 @@ import java.util.function.BiConsumer;
  */
 @Singleton
 @RequiresMetrics
+@TypeHint(value = {org.HdrHistogram.Histogram.class,org.HdrHistogram.ConcurrentHistogram.class})
 public class TimedInterceptor implements MethodInterceptor<Object, Object> {
 
     /**

